src/internal/parser.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  assertMapDocumentNode,
+  assertProfileDocumentNode,
+  EXTENSIONS,
+  MapDocumentNode,
+  ProfileDocumentNode,
+} from '@superfaceai/ast';
+import { parseMap, parseProfile, Source } from '@superfaceai/parser';
+import { createHash } from 'crypto';
+import { promises as fsp } from 'fs';
+import { join as joinPath } from 'path';
+
+import { Config } from '../config';
+
+export class Parser {
+  private static mapCache: Record<string, MapDocumentNode> = {};
+  private static profileCache: Record<string, ProfileDocumentNode> = {};
+
+  static async parseMap(
```

Comment:
Could we have some sort of flag which would force parsing of the map (profile)? In CLI check/lint/test it would be cool to actually compile on every run to ensure correct results 🤷

Reply:
sure

Reply:
After looking at it more I think that my idea is stupid 😅 - if we hash the source we will recompile on every source change anyway. 🤷

Reply:
true